src/wp-settings.php

```diff
@@ -110,6 +110,7 @@
 require ABSPATH . WPINC . '/class-wp-token-map.php';
 require ABSPATH . WPINC . '/formatting.php';
 require ABSPATH . WPINC . '/meta.php';
+require ABSPATH . WPINC . '/class-wp-exception.php';
```

Comment:
Could this class be used earlier? Any harm in moving it up to "Include files required for initialization.` section, such as after `class-wp-fatal-error-handler.php`?

Reply:
> Could this class be used earlier?

It currently isn't, but yes, in the future, it could be.

> Any harm in moving it up to "Include files required for initialization." section, such as after `class-wp-fatal-error-handler.php`?

No harm at all and in that case, maybe it should even be _before_ `class-wp-fatal-error-handler.php`, in case that class wants to use the exception (or handle it in some way).

Having said that, the current loading order is based on when it is _needed_ (at this time) and is still "early", i.e. in the `"// Load early WordPress files."` section, so this is _before_ WPDB is initialized, before translations are loaded, before plugins are loaded etc.

Maybe the "move it even earlier" should only happen if and when the need arises for the code to be loaded earlier ? That would make the case for _why_ it should be loaded earlier directly related to the change which created the _need_ for it to be loaded earlier. Does that make sense ?

Reply:
>No harm at all and in that case, maybe it should even be before `class-wp-fatal-error-handler.php`, in case that class wants to use the exception (or handle it in some way)

I agree.

>Maybe the "move it even earlier" should only happen if and when the need arises for the code to be loaded earlier ? That would make the case for why it should be loaded earlier directly related to the change which created the need for it to be loaded earlier. Does that make sense ?

It does, but ... 😉 

Thinking it makes more sense to put it with the fatal error handler, while making it available earlier as you noted.

I don't think there's an immediate cognitive correlation of its current placement to why it's there. But putting it before the fatal error handler could (maybe) make that correlation. What do I mean? Well, when I read its location, I wondered: hmm, what if the classes / code above needed it? "Grouping" it with the fatal error handler in those initialization file loads caught my attention.

As you noted, there's no harm in it being in either spot.

Reply:
I'll relocate it in the commit itself. No need to change it here.

Reply:
> I don't think there's an immediate cognitive correlation of its current placement to why it's there.

Well, there is a correlation. The function which uses the new class is defined in the `functions.php` file, which is why the class file is included _right before_ the `functions.php` file, but yes, that may not be an obvious connection for the casual onlooker.

> Thinking it makes more sense to put it with the fatal error handler, while making it available earlier as you noted.

Sounds good to me.